### DISCUSSION TITLE
STUN server connection

### DISCUSSION THREAD

**Discussion by bengoldenberg** [hyperjumptech/monika#614] [Q&A]
why do i need to connect to a STUN server? and how can i disable this feature? i dont want to allow external traffic from this component.

**Comment by dennypradipta** [comment]
Hi @bengoldenberg ,

Quoted from #310: "There are times when the computer where Monika is running is having a connection problem. When this happens, Monika will report and store this as an incident in the database even though the probed target probably has no issue."

That's why we implemented STUN server connection so that Monika could have a separate cycle to check if it can connect to the network by connecting to a STUN server (which in our case, is Google's STUN server)

**Reply by dennypradipta** [reply]
As for disabling STUN checks, currently we don't have a flag for it. But you could submit a PR if you want to :wink:

**Reply by amitde69** [reply]
we will submit a PR thanks :)

**Comment by sapiderman** [comment]
Hi @amitde69, @bengoldenberg,

You can disable STUN with `monika -s 0` to disable repeat lookups (it actually looks it up once only). In anycase, you can blocked STUN lookups do not break the probing loops, you'll only get a warning. Let us know if you have any feedback!
